src/datatypes.ts

```diff
@@ -153,6 +156,110 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * Time type for time data type attributes
+ */
+export type Time = {
+  hour: number;
+  minute: number;
+  second: number;
+  millisecond?: number;
+  timezoneHourOffset?: number;
+  timezoneMinuteOffset?: number;
+};
+
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value` in UTC.
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ */
+export function serializeTime(value: Time): string {
+  const utcHours =
+    value.hour + (value.timezoneHourOffset ? value.timezoneHourOffset : 0);
+  const utcMinutes =
+    value.minute +
+    (value.timezoneMinuteOffset ? value.timezoneMinuteOffset : 0);
+
+  let millisecondString;
+  // get millisecond into three digits
+  if (value.millisecond) {
+    if (value.millisecond < 10) {
+      millisecondString = "00" + value.millisecond;
+    } else if (value.millisecond < 100) {
+      millisecondString = "0" + value.millisecond;
+    } else {
+      millisecondString = value.millisecond;
+    }
+  }
```

Comment:
Looking at [the docs](https://www.w3.org/TR/xmlschema-2/#dateTime-lexical-representation), as I understand it the milliseconds don't actually need to be of a particular length, as opposed to the hour, minute and second representations.

Reply:
This was added so that if a user were to create a time as such:
time = {
...
millisecond: 42
}
It would be displayed as 0.42 instead of correctly storing it as 0.042. We need some kind of correction mechanism, but if there's no fixed length, I'll just use a division by 1000.

Reply:
Possibly just a division will work

Reply:
Aren't milliseconds inherently limited to 999? If we want there to be no fixed length then maybe we need a better naming convention or an option for more specificity.

Reply:
Hmm, so I was thinking that `0.42` was the correct value, but I now think I might indeed be misinterpreting it:

>  All ·minimally conforming· processors ·must· support (...) a minimum fractional second precision of milliseconds or three decimal digits (i.e. s.sss)

High school maths is a while ago, but it might indeed be the case that you need those leading zeroes to preserve the precision? Either way I don't think there's real harm in prepending them, so let's leave them in.

There's not necessarily a fixed length, it seems, but it's OK to have one:

> ·minimally conforming· processors ·may· set an application-defined limit on the maximum number of digits they are prepared to support in these two cases, in which case that application-defined maximum number ·must· be clearly documented.